src/classes/RD2_ScheduleService.cls

```diff
@@ -480,6 +480,9 @@ public without sharing class RD2_ScheduleService {
         if (schedule.DayOfMonth__c == RD2_Constants.DAY_OF_MONTH_LAST_DAY && schedule.InstallmentPeriod__c == RD2_Constants.INSTALLMENT_PERIOD_MONTHLY) {
             adjusted = Date.newInstance(adjusted.year(), adjusted.month(), Date.daysInMonth(adjusted.year(),adjusted.month()));
         }
+        else if (Schedule.DayOfMonth__c != null && Integer.valueOf(schedule.DayOfMonth__c) > Date.daysInMonth(firstValid.year(), firstValid.month())) {
```

Comment:
Beside moving the else up to the line with the bracket, I think the operator on this incorrect. If `DayofMonth__c` is more than the number of days in that month, then creating a date with that value should fail. I think the operation should be less-than-or-equal.

```suggestion
        } else if (schedule.DayOfMonth__c != null && Integer.valueOf(schedule.DayOfMonth__c) <= Date.daysInMonth(firstValid.year(), firstValid.month())) {
```

Reply:
Comment is outdated.

Reply:
Refactored that code, so please review newest.  Thanks!